src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -79,6 +80,16 @@ def __init__(self, use_case_id: UseCaseKey, outer_message: Message[MessageBatch]
 
         self._extract_messages()
 
+    def _should_index_tag_values(self) -> bool:
+        """
+        Helper function to determine if we should index tag values or not
+        """
+        if self.use_case_id == UseCaseKey.PERFORMANCE and not options.get(
+            "sentry-metrics.performance.index-tag-values"
+        ):
+            return False
+        return True
```

Comment:
I think the decision should be part of the indexer configuration. https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_metrics/configuration.py.
There is where we allow indexer specific configuration. WHether to index tag values or not seem to fit as indexer configuration. THe configuration can carry the option name.

Reply:
If we put this in the indexer configuration, that means we cannot dynamically change the parameter while the consumers are still running. We would rely on sentry.conf.py versus options.py. Is that something we are ok with? 

Also, if it goes in configuration.py, I think we would have to move towards using sentry.conf.py because options are not pickleable and iiuc everything in the configuration has to be pickleable. Correct me if I am wrong

Reply:
> If we put this in the indexer configuration, that means we cannot dynamically change the parameter while the consumers are still running. 

We can still achieve that. You can put the option name for the rollout in the config and then still rely on the option. So you still ahve the code that works in a way that is agnostic to the `UseCaseKey`